src/main/java/uk/gov/ons/census/casesvc/service/CaseReceipter.java

```diff
@@ -0,0 +1,27 @@
+package uk.gov.ons.census.casesvc.service;
+
+import static uk.gov.ons.census.casesvc.utility.QuestionnaireTypeHelper.iscontinuationQuestionnaireTypes;
+
+import org.springframework.stereotype.Component;
+import uk.gov.ons.census.casesvc.model.entity.Case;
+import uk.gov.ons.census.casesvc.model.entity.UacQidLink;
+
+@Component
+public class CaseReceipter {
```

Comment:
Is there a reason this can't live in the ReceiptService?

Reply:
It could, I was in favour of sticking it in the ReceiptService to start with.  However when I started on it , it felt nicer to stick it in it's own clean class.  This keeps a cleanish pattern, where both the ReceiptService & QuestionnaireLinkedService follow the same pattern lightweight pattern.  

The 'CaseReceipter' will be easier to test for both standard receipts and linking events, otherwiseI you'd want to test it through the public 'processReceipt' which would call the 'handleReceipting' function,  which is a bit more effort setting up more mocks and messages for each combination.  Then would you have to retest the handleReceipting for direct calls from the Linker?